rllib/models/torch/torch_distributions.py

```diff
@@ -48,17 +48,21 @@ def kl(self, other: "Distribution") -> TensorType:
     def sample(
         self,
         *,
-        sample_shape=torch.Size(),
+        sample_shape=None,
```

Comment:
why? what was wrong with the previous code?

Reply:
Sorry, should have left a comment.
This surfaced when importing TorchRLModule in RolloutWorker.
We can't have any framwork-specific calls that result from imports alone.
So incase someone does not have torch installed, if some random chain of imports ends up importin torch_distributions, they will fail if we leave torch.Size() here.

Reply:
This is because the TorchModule is not supposed to be imported in places that are framework agnostic (meaning that they should work for both tf or torch without having both installed). I will provide a suggestion on the rollout_worker code to solve this.

Reply:
https://github.com/ray-project/ray/pull/34640#discussion_r1192706003

Reply:
You should be able to to import anything in RLlib without getting missing framework errors.
If we import TorchRLModule somewhere (for example in RolloutWorker to test if an RLModule is a TorchRLModule), we also import torch distributions and therefore this would error out if torch was not installed.

Reply:
It only makes sense to call the TorchRLModule.compile() method here if an RLModule is at hand.
So why not check this? What you suggest is not a clean solution imo. You are testing a proxy.

I agree that RW should be framework agnostic, but we should not do this by getting rid of framework-specific imports when writing framework-specific code like in this case.
That's what we have try_import_torch and try_import_tf for after all.

To add to this, even if this gets rid of the `torch.Size()` import today, it will bite us back tomorrow because it will get imported somewhere else. And if not by us, it will be imported by users. Having torch.Size() there creates complications while benefiting no one.

Reply:
users who import this will have torch installed :) 

My check relies on this assumption that if RLModule is not None and Framework is torch it should be of type TorchRLModule, otherwise something is wrong.

Reply:
I understand, but based on the arguments I listed above I still think it's the wrong way to go.
I'd argue that you should not make assumptions if you don't need to.
Also, I guarantee that there will be import paths that users end up with that include these torch distributions without having torch installed at some point that it will frustrate them and result in an issue 😄

I'll disagree and commit to your solution here because we have to move forward but I'll knock on your door smirking when we run into an issue here lol